soco/core.py

```diff
@@ -1490,6 +1494,70 @@ def get_item_album_art_uri(self, item):
         else:
             return None
 
+    @only_on_master
+    def set_sleep_timer(self, sleep_time_seconds):
+        """Sets the sleep timer.
+
+        Args:
+            sleep_time_seconds (`int/NoneType`): How long to wait before
+                turning off speaker in seconds, None to cancel a sleep timer.
+                Maximum value of 86399
+
+        Raises:
+            SoCoException: Upon errors interacting with Sonos controller
+            ValueError: Argument/Syntax errors
+
+        """
+        # Note: A value of None for sleep_time_seconds is valid, and needs to
+        # be preserved distinctly separate from 0. 0 means go to sleep now,
+        # which will immediately start the sound tappering, and could be a
+        # useful feature, while None means cancel the current timer
+        if sleep_time_seconds == 0:
+            sleep_time = '00:00:00'
+        elif not sleep_time_seconds:
+            sleep_time = ''
+        elif sleep_time_seconds != int(sleep_time_seconds):
+            raise ValueError('invalid sleep_time_seconds, must be integer \
+                value between 0 and 86399 inclusive')
+        else:
+            sleep_time = str(
```

Comment:
I'm really not sure. One issue however is that I don't **think** that the str representation is guaranteed to remain the same. I think a more reliable way to do it, is to rely on the timedelta objets ability to format itself:
```python
sleep_time = format(datetime.timedelta(seconds=int(sleep_time_seconds)))
```

Maybe see if that gets rid of the error. In any case I think it is better, since we expect unicode everywhere internally in SoCo and this will in Python 2.7 convert convert to Python 2.7 str which is not unicode.